department-of-veterans-affairs/caseflow
# Import some realistic appeals

### Description
Implements [CASEFLOW-1638](https://vajira.max.gov/browse/CASEFLOW-1638). The _overwhelming_ majority of the >5,000 lines are four JSON files.

We want to import some realistic appeals ready for substitution from prod.

I used the [SanitizedJsonImporter](https://github.com/department-of-veterans-affairs/caseflow/wiki/Exporting-and-Importing-Appeals#importing) directly from seeds. This might make it easy to build upon later for other things, too.

So really this accomplishes two things:
* The direct goal of the story, to sanitize these 4 appeals and pull them in as test/demo data, and
* Start to build out a framework for importing more stuff this way in the future.

### Acceptance Criteria
- [ ] Seeding your database with this patch does not result in errors
- [ ] After seeding your database, you can find these appeals, e.g., appeal with UUID 9c14a2fd-348e-44d0-9465-5c3c6303b52d or a veteran with the (fictionalized/sanitized) file number of 000895588.
- [ ] You agree that the .json seeds are free of PII/PHI. See below for a bit more.

### Testing Plan
1. Check out this branch/PR
2. Reseed your database
3. Verify that some of the content added by the JSON seed files is now present in your database. Note that the ones with docket_type of hearing will (by design) not show a "Substitute Appellant" button yet.
4. Review the JSON files and confirm that they do not contain PII or PHI. See [the PII Handbook](https://github.com/department-of-veterans-affairs/appeals-team/blob/master/caseflow-team/0-how-we-work/pii-handbook.md#what-is-pii) (not public). These files _do_ show values for SSN, file number, name, and email, but they are fictional, generated randomly by the Faker gem. I encourage you to confirm that. Participant ID, or internal Caseflow IDs (e.g., UUIDs) are _not_ PII per that document.

(I did check myself that no PII had leaked before downloading it in the first place, but I strongly, strongly encourage you to double check.)

### User Facing Changes
TK

 BEFORE|AFTER
 ---|---

### Code Documentation Updates
- [ ] After this merges, or is at least looking good to go, I should update the [Exporting and Importing Appeals](https://github.com/department-of-veterans-affairs/caseflow/wiki/Exporting-and-Importing-Appeals) wiki page to reference the new ability to import them as seed data.
